src/dowhen/util.py

```diff
@@ -14,18 +14,24 @@
 from .types import IdentifierType
 
 
-def getrealsourcelines(obj) -> tuple[list[str], int]:
+def getrealsourcelines(
+    obj: CodeType | FunctionType | MethodType | ModuleType | type,
+) -> tuple[list[str], int]:
     try:
         lines, start_line = inspect.getsourcelines(obj)
-        # We need to find the actual definition of the function/class
-        # when it is decorated
-        while lines[0].strip().startswith("@"):
-            # If the first line is a decorator, we need to skip it
-            # and move to the next line
-            lines.pop(0)
-            start_line += 1
+
+        # If the first line is a decorator, find the next line
+        # that starts with "def ", "async def ", or "class ".
+        if lines and lines[0].strip().startswith("@"):
```

Comment:
What's the case when `lines` is empty? For every new check you added, I'd like a test to cover it.